src/test/lib/storage/encoded_segment_test.cpp

```diff
@@ -437,30 +437,48 @@ TEST_F(EncodedSegmentTest, SegmentReencoding) {
   auto value_segment = _create_int_with_null_value_segment(_row_count(EncodingType::FrameOfReference));
 
+  // Dictionary
   auto encoded_segment =
       this->_encode_segment(value_segment, DataType::Int,
                             SegmentEncodingSpec{EncodingType::Dictionary, VectorCompressionType::FixedSizeByteAligned});
   EXPECT_EQ(get_segment_encoding_spec(encoded_segment),
             (SegmentEncodingSpec{EncodingType::Dictionary, VectorCompressionType::FixedSizeByteAligned}));
   EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
+  encoded_segment = this->_encode_segment(
+      value_segment, DataType::Int, SegmentEncodingSpec{EncodingType::Dictionary, VectorCompressionType::SimdBp128});
+  EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
+  encoded_segment =
+      this->_encode_segment(value_segment, DataType::Int,
+                            SegmentEncodingSpec{EncodingType::Dictionary, VectorCompressionType::FixedSizeBitAligned});
+  EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
+
+  // RunLength
   encoded_segment = this->_encode_segment(value_segment, DataType::Int, SegmentEncodingSpec{EncodingType::RunLength});
   EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
+
+  // FrameOfReference
   encoded_segment =
       this->_encode_segment(value_segment, DataType::Int,
                             SegmentEncodingSpec{EncodingType::FrameOfReference, VectorCompressionType::SimdBp128});
   EXPECT_EQ(get_segment_encoding_spec(encoded_segment),
             (SegmentEncodingSpec{EncodingType::FrameOfReference, VectorCompressionType::SimdBp128}));
   EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
+  encoded_segment = this->_encode_segment(
+      value_segment, DataType::Int,
+      SegmentEncodingSpec{EncodingType::FrameOfReference, VectorCompressionType::FixedSizeByteAligned});
+  EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
+  encoded_segment = this->_encode_segment(
+      value_segment, DataType::Int,
+      SegmentEncodingSpec{EncodingType::FrameOfReference, VectorCompressionType::FixedSizeBitAligned});
+  EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
+
+  // LZ4
   encoded_segment =
       this->_encode_segment(value_segment, DataType::Int,
                             SegmentEncodingSpec{EncodingType::LZ4, VectorCompressionType::FixedSizeByteAligned});
   EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
   encoded_segment = this->_encode_segment(
-      value_segment, DataType::Int, SegmentEncodingSpec{EncodingType::Dictionary, VectorCompressionType::SimdBp128});
-  EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
-  encoded_segment = this->_encode_segment(
-      value_segment, DataType::Int,
-      SegmentEncodingSpec{EncodingType::FrameOfReference, VectorCompressionType::FixedSizeByteAligned});
+      value_segment, DataType::Int, SegmentEncodingSpec{EncodingType::LZ4, VectorCompressionType::FixedSizeBitAligned});
```

Comment:
Surprised we don't catch that. I think we fixed LZ4 compression to SimdBp128.

Reply:
as you can see above, the same test exist for LZ4 + FSByteA. Just let me know if I should remove both LZ4 + FSByteA and LZ4 + FSBitA or none.
All other error checks are not in scope of this PR I'd say.